### ISSUE TITLE
Better /ps list

### ISSUE THREAD

**Issue by rmalca** [espidev/ProtectionStones#251]
Hello, I know that with the command "/ps list" the protections of the player are shown but I would like it to be something "cleaner", for example on a server that I visit it is listed in this way (from unknown plugin):

![image](https://user-images.githubusercontent.com/3759745/113528919-a7024700-9587-11eb-8098-c9ec94996335.png)

Note that the first column is the name of the protection block, the second column is the distance to the block and the third is the coordinates). In this way it would be more understandable for the players and easier to find a placed protection block.

**Comment by espidev** [comment]
The main issue is that by default protection regions are not named, but they have ids (which are already displayed)

**Comment by rmalca** [comment]
Yes, you are right. I haven't done much research but it occurs to me that maybe IMHO we can identify the block and compare data with the "type" of the blocks in the configuration, this way we could use the "alias" in the first column. The distance could be calculated with the worldedit api and the coordinates as well.

**Comment by espidev** [comment]
The thing is I have to always keep legacy compatibility in mind, so that I don't all of a sudden get a hundred complaints about me randomly changing messages.

But the idea does sound good. My hands are just kind of tied on changing features like this, hmm not sure what the best way to proceed is...

It's possible that I could do it in 2.9.0 but that release is likely months from now

**Comment by espidev** [comment]
I'd recommend if you have some coding experience, it's possible to easily make a small plugin that provides a list feature that works for you. I can guide you on the Discord server with using the ProtectionStones API

**Comment by rmalca** [comment]
That sounds good, how could I start?

**Comment by espidev** [comment]
I do not have time to guide you through in learning how to make Spigot plugins (so you should search tutorials on that first), but once you have the plugin setup, add this to your build configuration: https://github.com/espidev/ProtectionStones/wiki/API

**Comment by rmalca** [comment]
That's fine thanks. I hope if I can finish with this plugin maybe one day it can be integrated with ProtectionStones.

**Comment by espidev** [comment]
From the documentation: https://jdps.espi.dev/

You can obtain a `PSPlayer` by doing `PSPlayer.fromPlayer(player)`. Then you can obtain their list of `PSRegion`s by having `psplayer.getPSPlayer(world, canBeMember)` replacing world with the bukkit world and canBeMember with a boolean.

**Comment by espidev** [comment]
> That's fine thanks. I hope if I can finish with this plugin maybe one day it can be integrated with ProtectionStones (within the repository).

Possibly, but it's more likely I'd have to end up doing it on my own since I have to worry about lots of internal stuff in ProtectionStones, along with allowing for translations

**Comment by rmalca** [comment]
> From the documentation: https://jdps.espi.dev/
> 
> You can obtain a `PSPlayer` by doing `PSPlayer.fromPlayer(player)`. Then you can obtain their list of `PSRegion`s by having `psplayer.getPSPlayer(world, canBeMember)` replacing world with the bukkit world and canBeMember with a boolean.

I think he was referring to "getPSRegions".

Perfect, that gives me a starting point.

**Comment by espidev** [comment]
:facepalm: yes, you're right lol
